src/Microsoft.AspNet.Mvc.Razor/Compilation/CompilerCache.cs

```diff
@@ -52,38 +57,50 @@ public CompilerCache([NotNull] IFileProvider fileProvider)
             [NotNull] string relativePath,
             [NotNull] Func<RelativeFileInfo, CompilationResult> compile)
         {
-            var normalizedPath = NormalizePath(relativePath);
             CompilerCacheResult cacheResult;
-            if (!_cache.TryGetValue(normalizedPath, out cacheResult))
+            if (!_cache.TryGetValue(relativePath, out cacheResult))
```

Comment:
If this is the "fast path lookup" for the case where `NormalizePath()` returns the same `string`, may be worth a code comment.